src/tiled/tilepainter.cpp

```diff
@@ -68,7 +70,7 @@ Cell TilePainter::cellAt(int x, int y) const
     const int layerX = x - mTileLayer->x();
     const int layerY = y - mTileLayer->y();
 
-    if (!mTileLayer->contains(layerX, layerY))
+    if (!mTileLayer->contains(layerX, layerY) && !mMapDocument->map()->infinite())
```

Comment:
Check is not necessary at all anymore. If the layer does not contain this position, it will anyway return an empty cell.